extensions-contrib/parquet-extensions/src/main/java/io/druid/data/input/parquet/ParquetHadoopInputRowParser.java

```diff
@@ -56,15 +61,50 @@ public ParquetHadoopInputRowParser(
     }
   }
 
+  private static LogicalType _timestampSpecLogicalType = null;
+
+  private LogicalType determineTimestampSpecLogicalType(Schema schema, String timestampSpecFields)
+  {
+    for (Schema.Field field : schema.getFields()) {
+      if (field.name().equals(timestampSpecFields)) {
+        return field.schema().getLogicalType();
+      }
+    }
+    // The timestamp field does not exists in the provided schema
+    return null;
+  }
+
   /**
    * imitate avro extension {@link AvroStreamInputRowParser#parseGenericRecord(GenericRecord, ParseSpec, List, boolean, boolean)}
    */
   @Override
   public InputRow parse(GenericRecord record)
   {
+    LogicalType logicalType;
+
+    // We don't want to determine the type every time if we already done so
+    if (null != _timestampSpecLogicalType) {
+      logicalType = _timestampSpecLogicalType;
+    } else {
+      logicalType = determineTimestampSpecLogicalType(record.getSchema(), timestampSpec.getTimestampColumn());
+    }
+
     GenericRecordAsMap genericRecordAsMap = new GenericRecordAsMap(record, false, binaryAsString);
-    TimestampSpec timestampSpec = parseSpec.getTimestampSpec();
-    DateTime dateTime = timestampSpec.extractTimestamp(genericRecordAsMap);
+
+    // Parse time timestamp based on the parquet schema.
+    // https://github.com/Parquet/parquet-format/blob/master/LogicalTypes.md#date
+    DateTime dateTime;
+    if (logicalType instanceof LogicalTypes.Date) {
+      int daysSinceEpoch = (Integer) genericRecordAsMap.get(timestampSpec.getTimestampColumn());
+
+      // days since epoch
+      long DAY_IN_MILISECONDS = 1000L * 60L * 60L * 24L;
+      dateTime = new DateTime(daysSinceEpoch * DAY_IN_MILISECONDS);
```

Comment:
Please ensure that the same time zone is used as returned from `timestampSpec.extractTimestamp()`